### ISSUE TITLE
Edge case: converting `foo === null || foo === undefined` with `foo == null` is not safe

### ISSUE THREAD

**Issue by jhpratt** [terser-js/terser#408]
**Bug report**

**Version**: 3.17.0

**Complete CLI command or `minify()` options used**

**`terser` input**:

```
terser -c <<< "const foo = document.all; console.log(foo === null || foo === undefined);"
```

**`terser` output or error**

```javascript
const foo=document.all;console.log(null==foo);
```

**Expected result**

```
const foo=document.all;console.log(null===foo||null===void 0);
```

`document.all` returns an `HTMLAllCollection`, which per the [WHATWG specification](https://html.spec.whatwg.org/multipage/common-dom-interfaces.html#the-htmlallcollection-interface), matches `== null`

> The Abstract Equality Comparison algorithm, when given objects implementing the `HTMLAllCollection` interface, returns true when compared to the `undefined` and `null` values. (Comparisons using the Strict Equality Comparison algorithm, and Abstract Equality comparisons to other values such as strings or objects, are unaffected.)

**Comment by fabiosantoscode** [comment]
This is why people hate javascript lol

**Comment by fabiosantoscode** [comment]
I'm not very happy to change Terser's behaviour here, but someone might hit this. I'm going to add the help wanted label. I'm looking at you @rogeriopvl. This is a good first PR for you.

**Comment by jhpratt** [comment]
@fabiosantoscode I agree it probably isn't a good idea to change the behavior, given that basically no one writes code using `document.all` nowadays, let alone needs to minify it.

This issue was really more of documentation, but also putting it out there if someone actually bothers to track usage throughout a program.

**Comment by fabiosantoscode** [comment]
I agree that nobody uses document.all nowadays. I started writing javascript using raw DOM, and even if I stayed that way for several lifetimes, I would have never found a use for document.all. I've changed my mind. I don't think this should warrant an increased file size for everything going through Terser checking for both null and undefined. I don't even think this is worth documenting.

If people want to read about trivia from ancient times they can read something else that isn't a piece of documentation.

Some things are best left forgotten.

Thanks for your report though :)
